src/config.rs

```diff
@@ -1,19 +1,48 @@
 //! All sync client configuration structs and logic
 use std::{
-	fmt::Display,
-	path::{Path, PathBuf},
+	ops::{Deref, DerefMut},
+	path::Path,
 };
 
 use anyhow::{bail, Result};
-use ldap_poller::{config::TLSConfig, AttributeConfig, CacheMethod, ConnectionConfig, Searches};
 use serde::Deserialize;
 use url::Url;
 
+use crate::{
+	sources::{csv::CsvSourceConfig, ldap::LdapSourceConfig, ukt::UktSourceConfig},
+	zitadel::ZitadelConfig,
+};
+
 /// App prefix for env var configuration
 const ENV_VAR_CONFIG_PREFIX: &str = "FAMEDLY_LDAP_SYNC";
 /// Separator for setting a list using env vars
 const ENV_VAR_LIST_SEP: &str = " ";
 
+/// The main sync tool with all configurations
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub struct Config {
+	/// Configuration related to Zitadel provided by Famedly
+	pub zitadel_config: ZitadelConfig,
```

Comment:
There are quite a lot of backwards incompatible changes to the configuration format. This is fine, but needs to be communicated clearly to Infra. To ensure it doesn't get lost, please mark a relevant commit as breaking with a `!` and include some information in the long commit message. For example, something like:

```
refactor!: Adjust config struct for multi-source trait

BREAKING CHANGE: The configuration format has changed significantly and
is not backwards compatible. Please refer to the sample configuration for
an up-to-date example. Notably, the `famedly` configuration struct is now
renamed to `zitadel_config`.

Reply:
In general the commit messages need a bit of work, probably some rebasing/squashing too.